stacks-signer/src/v0/signer.rs

```diff
@@ -617,6 +667,81 @@ impl Signer {
         }
     }
 
+    /// Check the current tracked submitted block proposal to see if it has timed out.
+    /// Broadcasts a rejection and marks the block locally rejected if it has.
+    fn check_submitted_block_proposal(&mut self) {
```

Comment:
I have a bad feeling about this.  Block validation can be slow in wall-clock time terms for a multitude of reasons (slow RPC, slow disk, memory pressure, etc.), but they're guaranteed to evaluate in all cases.  Wouldn't this cause a lot of unnecessary, unpredictable, hard-to-debug, hard-to-reproduce block rejections?

If the goal here is to compel miners to produce small blocks, then what if instead signers advertised a max compute budget per block that miners had to adhere to?  Stackers have access to a multitude of StackerDB slots that were heretofore reserved for DKG.  We could repurpose some of them to store a "signer policy" structure that, among other things, informed miners as to what the signer expects from the blocks they'll validate.

Reply:
Yes I agree. We were seeing behaviours where the miner was essentially dosing the signers node by producing way too many block proposals and then the queue of requests getting spammed too much such that the node could not even issue the 429 error we expect which prevents more than one block proposal being processed at once.

Reply:
Yeah, I think that's right.